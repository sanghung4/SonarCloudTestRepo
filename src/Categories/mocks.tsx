import { DocumentNode } from 'graphql';
import {
  ProductCategoriesDocument,
  ProductCategoriesQuery
} from 'generated/graphql';

export const success: MockCategory = {
  request: {
    query: ProductCategoriesDocument,
    variables: {
      engines: 'plumbing_hvac'
    }
  },
  result: {
    data: {
      productCategories: {
        categories: [
          {
            name: 'Faucets, Fixtures & Appliances',
            children: [
              {
                name: 'TBC',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Faucets, Showers & Bathroom Accessories',
                children: [
                  {
                    name: 'Showers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Bathroom Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'Tub & Shower',
                    __typename: 'Category'
                  },
                  {
                    name: 'Bathroom Faucets',
                    __typename: 'Category'
                  },
                  {
                    name: 'Kitchen Faucets',
                    __typename: 'Category'
                  },
                  {
                    name: 'Tub Filler',
                    __typename: 'Category'
                  },
                  {
                    name: 'Spare Parts',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Kitchen Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Appliances',
                children: [
                  {
                    name: 'Refrigerators And Freezers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Hoods and Vents',
                    __typename: 'Category'
                  },
                  {
                    name: 'Outdoor Grills and BBQ',
                    __typename: 'Category'
                  },
                  {
                    name: 'Ranges and Stoves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Microwaves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Garbage Disposals',
                    __typename: 'Category'
                  },
                  {
                    name: 'Appliance Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'Dishwashers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Dryers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Ovens',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Toilets, Urinals, Sinks & Flush Valves',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Toilet & Flush Valve Spare Parts',
                    __typename: 'Category'
                  },
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  },
                  {
                    name: 'Manual Flush Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Spare Parts',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Fans & Ventilation',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Tools',
                children: [],
                __typename: 'Category'
              }
            ],
            __typename: 'Category'
          },
          {
            name: 'Plumbing Installation, Tools, Hardware & Safety',
            children: [
              {
                name: 'Plumbing Installation',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Pipe Clamps',
                    __typename: 'Category'
                  },
                  {
                    name: 'Pipe Insulation',
                    __typename: 'Category'
                  },
                  {
                    name: 'Faucet and Toilet Parts',
                    __typename: 'Category'
                  },
                  {
                    name: 'Gaskets',
                    __typename: 'Category'
                  },
                  {
                    name: 'Pipe Hangers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Riser',
                    __typename: 'Category'
                  },
                  {
                    name: 'Insulation',
                    __typename: 'Category'
                  },
                  {
                    name: 'Anchors and Settings',
                    __typename: 'Category'
                  },
                  {
                    name: 'Compression Fittings',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Tools',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Power Tool',
                    __typename: 'Category'
                  },
                  {
                    name: 'Brushes',
                    __typename: 'Category'
                  },
                  {
                    name: 'Tool',
                    __typename: 'Category'
                  },
                  {
                    name: 'Shovels',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'TBC',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Hardware',
                children: [
                  {
                    name: 'Screws',
                    __typename: 'Category'
                  },
                  {
                    name: 'Washers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Nuts',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Bolts',
                    __typename: 'Category'
                  },
                  {
                    name: 'Pipe Hangers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Nails',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Safety',
                children: [
                  {
                    name: 'Gloves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Glasses',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Hard Hat',
                    __typename: 'Category'
                  },
                  {
                    name: 'Vest',
                    __typename: 'Category'
                  },
                  {
                    name: 'Shoe Covers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Beanie',
                    __typename: 'Category'
                  },
                  {
                    name: 'Ear Plugs',
                    __typename: 'Category'
                  },
                  {
                    name: 'Face Shield',
                    __typename: 'Category'
                  },
                  {
                    name: 'Knee Pads',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Tools, Hardware & Safety',
                children: [
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Brushes',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Appliances',
                children: [
                  {
                    name: 'Garbage Disposals',
                    __typename: 'Category'
                  },
                  {
                    name: 'Dishwashers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Washing Machines',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Pipe Accessories',
                children: [
                  {
                    name: 'Pipe Insulation',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'HVAC',
                children: [
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Discontinued',
                children: [],
                __typename: 'Category'
              }
            ],
            __typename: 'Category'
          },
          {
            name: 'Pipe & Fittings',
            children: [
              {
                name: 'Metal Pipe & Fittings',
                children: [
                  {
                    name: 'Copper Tube & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'Steel Pipe & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'Cast Iron Pipe & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'Brass Pipe & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'Ductile Iron Pipe & Fittings',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Plastic Pipe & Fittings - Supply',
                children: [
                  {
                    name: 'PEX Pipe & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'CPVC Pipe & Fittings',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Plastic Pipe & Fittings - Drainage',
                children: [
                  {
                    name: 'PVC Pipe & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'ABS Pipe & Fittings',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'TBC',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Plumbing Installation',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              }
            ],
            __typename: 'Category'
          },
          {
            name: 'Heating & Cooling',
            children: [
              {
                name: 'Air Distribution',
                children: [
                  {
                    name: 'Sheet Metal Pipe & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'Grills, Registers & Diffusers',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Flex Duct',
                    __typename: 'Category'
                  },
                  {
                    name: 'Sheet Metal Duct Pipe and Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'Access & Inspection Panels',
                    __typename: 'Category'
                  },
                  {
                    name: 'CEILING RADIATION DAMPERS',
                    __typename: 'Category'
                  },
                  {
                    name: 'Grills, Registers, & Diffusers',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Repair & Replacement Parts',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Capacitors',
                    __typename: 'Category'
                  },
                  {
                    name: 'OEM Parts',
                    __typename: 'Category'
                  },
                  {
                    name: 'Contactors, Relays, Transformers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Motors',
                    __typename: 'Category'
                  },
                  {
                    name: 'Driers & Sight Glasses',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'TBC',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Residential Equipment',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'AC Condensers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Residential Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'Air Handlers',
                    __typename: 'Category'
                  },
                  {
                    name: 'HP Condensers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Coils',
                    __typename: 'Category'
                  },
                  {
                    name: 'Furnaces',
                    __typename: 'Category'
                  },
                  {
                    name: 'Packaged Systems',
                    __typename: 'Category'
                  },
                  {
                    name: 'furnaces',
                    __typename: 'Category'
                  },
                  {
                    name: 'Condensers',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Commercial Equipment',
                children: [
                  {
                    name: 'Commercial Packaged Systems',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Commercial Air Handlers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Geothermal',
                    __typename: 'Category'
                  },
                  {
                    name: 'Packaged Systems',
                    __typename: 'Category'
                  },
                  {
                    name: 'Residential Accessories',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Installation Supplies',
                children: [
                  {
                    name: 'Tapes, Silicon, & Adhesives',
                    __typename: 'Category'
                  },
                  {
                    name: 'Electrical',
                    __typename: 'Category'
                  },
                  {
                    name: 'Insulation',
                    __typename: 'Category'
                  },
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  },
                  {
                    name: 'Condensate Management',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Fans & Ventilation',
                children: [
                  {
                    name: 'Gas Vent Pipe & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Ventilation Fans',
                    __typename: 'Category'
                  },
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'IAQ',
                children: [
                  {
                    name: 'Air Filters',
                    __typename: 'Category'
                  },
                  {
                    name: 'Air Cleaners',
                    __typename: 'Category'
                  },
                  {
                    name: 'Dehumidifiers',
                    __typename: 'Category'
                  },
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  },
                  {
                    name: 'Humidifiers',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Controls, Sensors, & Zoning',
                children: [
                  {
                    name: 'Thermostats',
                    __typename: 'Category'
                  },
                  {
                    name: 'Zoning',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Ductless',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Condensers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Air Handlers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Residential Accessories',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              }
            ],
            __typename: 'Category'
          },
          {
            name: 'Hot Water, Valves, Irrigation & Pumps',
            children: [
              {
                name: 'TBC',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Water Heater Parts & Accessories',
                children: [
                  {
                    name: 'Venting',
                    __typename: 'Category'
                  },
                  {
                    name: 'Water Heater Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'Commercial Conversion Kits',
                    __typename: 'Category'
                  },
                  {
                    name: 'Burners',
                    __typename: 'Category'
                  },
                  {
                    name: 'Heating Element',
                    __typename: 'Category'
                  },
                  {
                    name: 'Gas Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Switches',
                    __typename: 'Category'
                  },
                  {
                    name: 'Expansion Tanks',
                    __typename: 'Category'
                  },
                  {
                    name: 'Instantaneous',
                    __typename: 'Category'
                  },
                  {
                    name: 'Electronic Boards',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Specialty Plumbing Valves',
                children: [
                  {
                    name: 'Hydrants/Flow Control Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Pressure Reducing Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Wye Strainers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Valve Kits',
                    __typename: 'Category'
                  },
                  {
                    name: 'Relief Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Thermostatic Mixing Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Hose Bibbs',
                    __typename: 'Category'
                  },
                  {
                    name: 'Dielectric Unions',
                    __typename: 'Category'
                  },
                  {
                    name: 'Sillcocks',
                    __typename: 'Category'
                  },
                  {
                    name: 'Automatic Control Valves',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Import Valves',
                children: [
                  {
                    name: 'Ball Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Check Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Gate Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Balancing Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Butterfly Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Valve Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'Globe Valves',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Domestic Valves',
                children: [
                  {
                    name: 'Ball Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Check Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Butterfly Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Gate Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Valve Accessories',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Pumps',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Parts and Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  },
                  {
                    name: 'Recirculating Pumps',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Backflow Preventers & Accessories',
                children: [
                  {
                    name: 'Backflow Preventers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Backflow Preventer Vacuum Breakers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Backflow Preventer Kits',
                    __typename: 'Category'
                  },
                  {
                    name: 'Air Gaps',
                    __typename: 'Category'
                  },
                  {
                    name: 'Backflow Preventer Accessories',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Water Treatment',
                children: [
                  {
                    name: 'Water Filters',
                    __typename: 'Category'
                  },
                  {
                    name: 'Water Filter Parts & Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'Water Softener Parts & Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Water Softeners',
                    __typename: 'Category'
                  },
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  },
                  {
                    name: 'Water Conditioners',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Water Heaters - Residential',
                children: [
                  {
                    name: 'Gas',
                    __typename: 'Category'
                  },
                  {
                    name: 'Electric',
                    __typename: 'Category'
                  },
                  {
                    name: 'Heat Pump',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Water Heaters - Commercial',
                children: [
                  {
                    name: 'Electric',
                    __typename: 'Category'
                  },
                  {
                    name: 'Gas',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              }
            ],
            __typename: 'Category'
          },
          {
            name: 'TBC',
            children: [
              {
                name: 'TBC',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Sheet Metal Pipe & Fittings',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              }
            ],
            __typename: 'Category'
          },
          {
            name: 'Faucets, Fixtures Appliances',
            children: [
              {
                name: 'TBC',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              }
            ],
            __typename: 'Category'
          },
          {
            name: 'Faucets, FiXtures & Appliances',
            children: [],
            __typename: 'Category'
          }
        ],
        __typename: 'ProductCategories'
      },
      __typename: 'Query'
    }
  }
};

type MockCategory = {
  request: {
    query: DocumentNode;
    variables: {
      engines?: string;
    };
  };
  result: {
    data: ProductCategoriesQuery | undefined;
  };
};
