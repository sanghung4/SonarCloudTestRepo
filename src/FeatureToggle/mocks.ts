import { Feature } from 'generated/graphql';

export const mockFeatures: Feature[] = [
  { id: '1', isEnabled: false, name: 'noooo' },
  { id: '2', isEnabled: true, name: 'fine' },
  { id: '3', isEnabled: true, name: 'test' }
];
