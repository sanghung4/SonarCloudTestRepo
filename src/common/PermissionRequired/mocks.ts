import { UserContextType } from 'AuthProvider';
import { Permission } from 'common/PermissionRequired';

export const authConfigNoPerm = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
export const authConfig = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    userId: 'test',
    permissions: [Permission.EDIT_LIST, Permission.APPROVE_CART],
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
