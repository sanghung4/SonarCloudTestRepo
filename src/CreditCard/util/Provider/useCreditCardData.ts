import { useCheckoutContext } from './../../../Checkout/CheckoutProvider';
import { useContext, useState } from 'react';

import { omit } from 'lodash-es';

import {
  CreditCardState,
  defaultCreditCardContext
} from 'CreditCard/util/config';
import useCreditCardFn from 'CreditCard/util/Provider/useCreditCardFn';
import {
  CreditCard,
  useCreditCardElementInfoLazyQuery,
  useCreditCardListQuery,
  useCreditCardSetupUrlLazyQuery,
  useDeleteCreditCardMutation
} from 'generated/graphql';
import { useSelectedAccountsContext } from 'providers/SelectedAccountsProvider';
import { Permission, checkUserPermission } from 'common/PermissionRequired';
import { AuthContext } from 'AuthProvider';

export default function useCreditCardData() {
  /**
   * Custom Hooks
   */
  const { selectedAccounts } = useSelectedAccountsContext();
  const { profile } = useContext(AuthContext);
  const accountId = selectedAccounts.billTo?.erpAccountId ?? '';

  /**
   * Context
   */
  const { paymentData, setPaymentData, updateCart } = useCheckoutContext();
  const canManagePaymentMethods = checkUserPermission(profile, [
    Permission.MANAGE_PAYMENT_METHODS
  ]);

  /**
   * State
   */
  const [resUrl, setResUrl] = useState<string>();
  const [iframeUrl, setIframeUrl] = useState<string>();
  const [creditCard, setCreditCard] = useState<CreditCard>();
  const [creditCardData, setCreditCardData] = useState<CreditCard[]>();
  const [creditCardState, setCreditCardState] = useState<CreditCardState>(
    defaultCreditCardContext.creditCardState
  );
  const [getCreditCardLoading, setGetCreditCardLoading] = useState(false);
  const [parsingCCResponse, setParsingCCResponse] = useState(false);
  const [selectedCreditCard, setSelectedCreditCard] = useState(
    defaultCreditCardContext.selectedCreditCard
  );
  const [shouldSaveCreditCard, setShouldSaveCreditCard] = useState(false);
  const [updatingList, setUpdatingList] = useState(false);

  /**
   * Functions
   */
  const {
    addCreditCardLoading,
    deleteComplete,
    deleteError,
    elementInfoComplete,
    listComplete,
    setupUrlComplete
  } = useCreditCardFn({
    creditCardData,
    setCreditCard,
    setCreditCardData,
    setCreditCardState,
    setGetCreditCardLoading,
    setIframeUrl,
    setParsingCCResponse,
    setResUrl,
    setShouldSaveCreditCard,
    setUpdatingList,
    shouldSaveCreditCard,
    updateCartCreditCard
  });

  /**
   * GraphQL
   */
  const { loading: creditCardListLoading } = useCreditCardListQuery({
    skip: !selectedAccounts.billTo?.erpAccountId || !canManagePaymentMethods,
    variables: { accountId },
    onCompleted: listComplete
  });
  const [getCreditCardSetupUrl] = useCreditCardSetupUrlLazyQuery({
    fetchPolicy: 'no-cache',
    onCompleted: setupUrlComplete
  });
  const [getCreditCardElementInfo] = useCreditCardElementInfoLazyQuery({
    onCompleted: elementInfoComplete
  });
  const [deleteCreditCardMutation, { loading: deleteCreditCardLoading }] =
    useDeleteCreditCardMutation({
      refetchQueries: ['CreditCardList'],
      onCompleted: deleteComplete,
      onError: deleteError
    });

  /**
   * Defs
   */
  function updateCartCreditCard(creditCard?: CreditCard) {
    const newPaymentData = { ...paymentData };
    if (creditCard) {
      newPaymentData.creditCard = omit(creditCard, '__typename');
    }
    setPaymentData(newPaymentData);
    updateCart(newPaymentData);
  }

  /**
   * Function defs
   */
  async function deleteCreditCard(accountId: string, creditCardId: string) {
    setUpdatingList(true);

    await deleteCreditCardMutation({
      variables: {
        accountId,
        creditCardId
      }
    });
  }

  /**
   * Outout
   */
  return {
    resUrl,
    setResUrl,
    iframeUrl,
    setIframeUrl,
    creditCard,
    setCreditCard,
    creditCardData,
    setCreditCardData,
    creditCardState,
    setCreditCardState,
    getCreditCardLoading,
    setGetCreditCardLoading,
    parsingCCResponse,
    setParsingCCResponse,
    selectedCreditCard,
    setSelectedCreditCard,
    shouldSaveCreditCard,
    setShouldSaveCreditCard,
    updatingList,
    setUpdatingList,
    creditCardListLoading,
    getCreditCardSetupUrl,
    getCreditCardElementInfo,
    addCreditCardLoading,
    deleteCreditCardMutation,
    deleteCreditCardLoading,
    updateCartCreditCard,
    deleteCreditCard
  };
}
