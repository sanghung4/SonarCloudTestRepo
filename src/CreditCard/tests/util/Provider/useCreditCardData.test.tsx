import { act } from '@testing-library/react';
import { CheckoutContext } from 'Checkout/CheckoutProvider';
import { mockCheckoutContext } from 'Checkout/tests/mocks/provider.mocks';
import { mockCreditCard } from 'CreditCard/tests/mocks/index.mocks';
import {
  mockCreditCardData,
  mockCreditCardFn
} from 'CreditCard/tests/mocks/useCreditCardData.mocks';
import useCreditCardData from 'CreditCard/util/Provider/useCreditCardData';
import { omit } from 'lodash-es';
import { render } from 'test-utils/TestWrapper';
import { mockSelectedAccounts } from 'hooks/tests/mocks/useSelectedAccounts.mocks';

/**
 * Mock values
 */
const mocks = {
  checkout: { ...mockCheckoutContext },
  deleteMutation: jest.fn(async () => new Promise((res) => res)),
  output: { loading: false, mutation: jest.fn() },
  selectedAccounts: { ...mockSelectedAccounts }
};

/**
 * Mock methods
 */
jest.mock('generated/graphql', () => ({
  ...jest.requireActual('generated/graphql'),
  useCreditCardListQuery: () => ({ loading: mocks.output.loading }),
  useCreditCardSetupUrlLazyQuery: () => [mocks.output.mutation],
  useCreditCardElementInfoLazyQuery: () => [mocks.output.mutation],
  useDeleteCreditCardMutation: () => [
    mocks.deleteMutation,
    { loading: mocks.output.loading }
  ]
}));
jest.mock('CreditCard/util/Provider/useCreditCardFn', () => ({
  ...jest.requireActual('CreditCard/util/Provider/useCreditCardFn'),
  __esModule: true,
  default: () => ({ ...mockCreditCardFn })
}));

/**
 * Mock function
 */
function setup(p: typeof mocks) {
  const output = { ...mockCreditCardData };
  function MockComponent() {
    Object.assign(output, useCreditCardData());
    return null;
  }
  render(
    <CheckoutContext.Provider value={p.checkout}>
      <MockComponent />
    </CheckoutContext.Provider>
  );
  return output;
}

/**
 * Test
 */
describe('CreditCard - util/Provider/useAddCreditCard', () => {
  afterEach(() => {
    mocks.checkout = { ...mockCheckoutContext };
    mocks.deleteMutation = jest.fn(async () => new Promise((res) => res));
    mocks.output = { loading: false, mutation: jest.fn() };
    mocks.selectedAccounts = { ...mockSelectedAccounts };
  });

  it('Expect to match snapshot of the output', () => {
    const result = setup(mocks);
    expect(result).toMatchSnapshot();
  });

  it('Expect `updateCartCreditCard` to call with no credit card', () => {
    const { updateCartCreditCard } = setup(mocks);
    updateCartCreditCard();
    expect(mocks.checkout.setPaymentData).toBeCalledWith({
      ...mocks.checkout.paymentData
    });
  });

  it('Expect `updateCartCreditCard` to call with credit card', () => {
    const { updateCartCreditCard } = setup(mocks);
    updateCartCreditCard(mockCreditCard);
    expect(mocks.checkout.setPaymentData).toBeCalledWith({
      ...mocks.checkout.paymentData,
      creditCard: omit(mockCreditCard, '__typename')
    });
  });

  it('Expect `deleteCreditCard` to be called', async () => {
    const { deleteCreditCard } = setup(mocks);
    await act(async () => await deleteCreditCard('', ''));
    await act(() => new Promise((res) => setTimeout(res, 0)));
    expect(mocks.deleteMutation).toBeCalled();
  });
});
