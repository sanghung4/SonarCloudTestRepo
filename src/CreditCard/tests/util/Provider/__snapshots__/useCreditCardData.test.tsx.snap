// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditCard - util/Provider/useAddCreditCard Expect to match snapshot of the output 1`] = `
Object {
  "addCreditCardLoading": false,
  "creditCard": undefined,
  "creditCardData": undefined,
  "creditCardListLoading": false,
  "creditCardState": 0,
  "deleteCreditCard": [Function],
  "deleteCreditCardLoading": false,
  "deleteCreditCardMutation": [MockFunction],
  "getCreditCardElementInfo": [MockFunction],
  "getCreditCardLoading": false,
  "getCreditCardSetupUrl": [MockFunction],
  "iframeUrl": undefined,
  "parsingCCResponse": false,
  "resUrl": undefined,
  "selectedCreditCard": "",
  "setCreditCard": [Function],
  "setCreditCardData": [Function],
  "setCreditCardState": [Function],
  "setGetCreditCardLoading": [Function],
  "setIframeUrl": [Function],
  "setParsingCCResponse": [Function],
  "setResUrl": [Function],
  "setSelectedCreditCard": [Function],
  "setShouldSaveCreditCard": [Function],
  "setUpdatingList": [Function],
  "shouldSaveCreditCard": false,
  "updateCartCreditCard": [Function],
  "updatingList": false,
}
`;
